sellercenter-category/src/main/java/com/gfg/sellercenter/category/service/CategoryService.java

```diff
@@ -0,0 +1,44 @@
+package com.gfg.sellercenter.category.service;
+
+import com.gfg.sellercenter.category.entity.CategoryWithProductInformation;
+import com.gfg.sellercenter.category.reader.HttpReader;
+import lombok.AllArgsConstructor;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@AllArgsConstructor
+public class CategoryService {
+
+    private final HttpReader httpReader;
+
+    public Map<Integer, CategoryWithProductInformation> getProductCategoriesByProductsIds(List<Integer> productIds) throws Exception {
+        return createSettingEntity(
+                this.httpReader.getProductCategoriesByProductIds(productIds)
+        );
+    }
+
+    private Map<Integer, CategoryWithProductInformation> createSettingEntity(JSONArray response) {
+        Map<Integer, CategoryWithProductInformation> categoryToProductMap = new HashMap<>();
+
+        response.forEach(
```

Comment:
I suggest to not do any mapping manually. 
```ObjectMapper mapper = new ObjectMapper();
mapper.readValue("input stream, string, file, URL - this method can work with just anything!", HereWhatEverClass);
```
can take url, get it, and automatically map to array (list) of objects of desired class.
After that list can me transformed to map. 
If you don't wan't to do it, or you don't like - you can leave it like it is. Most important: it works.

Reply:
hmm ... maybe makes sense... but i am not a big fan of to much magic :D
For this changes i need to annotate my entity with @JsonPropertyName right?

Reply:
Yes, this annotation should be added only in case if in raw JSON key does not match property name. We have couple examples (CreateConsignmentResponse one of them as I remember). And it was just a suggestion. Current code works fine as well.

Reply:
In general I would say it is always better to use object mapper

Reply:
What is better using something magical instead of using full control? I think it is as well kind of programming style of a person itself or?